governance/approveDeny/approveDeny.groovy

```diff
@@ -0,0 +1,232 @@
+            int index = 0
+
+            if (items.length == 4) {
+                product_name_space = items[0]
+                index++
+            }
+
+            def module_name = items[index]
+            def version = items[index + 1]
+
+            if (items[index + 2].compareTo("manifest.json") == 0) {
+                log.warn "ipServer: ${settings.getIpServer()}"
+                log.warn "entitlementServer: ${settings.getEntitlementServer()}"
+                log.warn "transaction_ip: ${transaction_ip}"
+                log.warn "app_name: ${app_name}"
+                log.warn "repository_name: ${repository_name}"
+                log.warn "module_name: ${module_name}"
+                log.warn "product_name_space: ${product_name_space}"
+                log.warn "version: ${version}"
+                log.warn "email: ${email}"
+
+                Result result = compliance.validate(settings, transaction_ip, app_name, repository_name, module_name, product_name_space, version, email)
+
+                if (result.success()) {
+                    status = result.status
+                }
+            }
+        }
+        catch(Exception e) {
+            log.warn e.toString()
+        }
+    }
+}
+
+executions {
+    test { params ->
+        log.warn(log.getClass().toString())
+        Result result = approve.validate(settings, "172.17.0.1", "test-repo", "admin", "null")
+
+        log.warn(String.valueOf(result.status))
+        log.warn(result.message)
+    }
+}
+
+final class Globals {
+    private Globals() {}
+
+    final static SUCCESS = "success"
+    final static FAILED = "failed"
+    final static ERROR = 'error'
+}
+
+class Settings {
+    Logger log
+    String ipServer
+    String entitlementServer
+
+    Settings(ctx, log) {
+        this.log = log
+        File confFile = new File("${ctx.artifactoryHome.getEtcDir()}/plugins", "approveDeny.json")
+        def reader
+
+        try {
+            reader = new FileReader(confFile)
+            JsonSlurper slurper = new JsonSlurper()
+            Map content = slurper.parse(reader)
+            assert content instanceof Map
+
+            this.ipServer = content.ipServer
+            this.entitlementServer = content.entitlementServer
+        } finally {
+            if (reader) {
+                reader.close()
+            }
+        }
+    }
+
+    String getIpServer() {
+        return this.ipServer
+    }
+
+    String getEntitlementServer() {
+        return this.entitlementServer
+    }
+}
+
+class ThreadSafeHTTPBuilder extends HTTPBuilder {
+    protected AbstractHttpClient createClient(HttpParams params) {
+        PoolingClientConnectionManager cm = new PoolingClientConnectionManager()
+        cm.setMaxTotal(200) // Increase max total connection to 200
+        cm.setDefaultMaxPerRoute(20) // Increase default max connection per route to 20
+        new DefaultHttpClient(cm, params)
+    }
+}
+
+class Result {
+    int status
```

Comment:
There is already an HTTP result class built in to RT.  Check preventUnapproved.groovy and use how it does results please.